nuxeo-services/nuxeo-platform-directory/nuxeo-platform-directory-mongodb/src/main/java/org/nuxeo/directory/mongodb/MongoDBSession.java

```diff
@@ -78,10 +78,10 @@
 
     protected String countersCollectionName;
 
-    public MongoDBSession(MongoDBDirectory directory) {
+    public MongoDBSession(MongoDBDirectory directory, MongoClient client) {
         super(directory, MongoDBReference.class);
         MongoDBDirectoryDescriptor desc = directory.getDescriptor();
-        client = MongoDBConnectionHelper.newMongoClient(desc.getServerUrl());
+        this.client = client;
```

Comment:
And even, don't use a local `client` field in `MongoDBSession`, but have a local `getClient` method that does `return ((MongoDBDirectory) directory).getClient()` and just use that instead of the `client` field.